node-notes-cli/app.js

```diff
@@ -0,0 +1,57 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+let data = await readFile('data.json', 'utf8');
+data = JSON.parse(data);
+const notesObj = data.notes;
+
+async function rewrite() {
+  await writeFile('data.json', JSON.stringify(data, null, 2));
+}
+
+async function read() {
+  for (const [key, value] of Object.entries(notesObj)) {
+    console.log(`${key}: ${value}`);
+  }
+}
+
+async function add() {
+  const id = data.nextId;
+  data.nextId++;
+  notesObj[id] = process.argv[3];
+  rewrite();
+}
+
+async function poof() {
+  if (notesObj[process.argv[3]]) {
+    delete notesObj[process.argv[3]];
+    rewrite();
+  } else {
+    throw new Error('This ID does not exist!');
```

Comment:
It usually makes for more readable code if you detect the error condition at the top of the function and exit the function, then move on with the "happy path". For comparison:
```
async function poof() {
  const id = process.argv[3];
  if (!notesObj[id]) {
    throw new Error(`ID ${id} does not exist!`);
  }
  delete notesObj[id];
  rewrite();
}    
```
(Note that I also report the ID it was looking for in the error message, which will help the user better understand what happened.)